### PR TITLE
fix(client-plugins): default description, instructions, and seed to empty string 

### PR DESCRIPTION
**Pull request by huyenltnguyen**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

I got some warning messages in the terminal regarding potentially broken curriculum files, but the challenge titles were just "Task 1" and "Task 2", so I couldn't tell which files they are.

This PR:
- Includes block name in the message to help make it clearer
- Fixes the curriculum files in question

<details>
  <summary>Screenshot</summary>

  | Before | After |
  | --- | --- |
  | Screenshot 2024-10-15 at 03 03 21 | Screenshot 2024-10-15 at 03 08 14 |
</details>

### CODE REVIEW

File: curriculum/challenges/english/24-b1-english-for-developers/learn-how-to-describe-places-and-events/66a909dc2e382527ead44c23.md [hunk 1]
```diff
@@ -4,9 +4,10 @@ title: Task 1
 challengeType: 19
 dashedName: task-1
 ---
+
 <!-- (Audio) Bob: Hey Lisa, do you have a minute?  -->
 
-# --instructions--
+# --description--
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/24-b1-english-for-developers/learn-how-to-describe-places-and-events/66a909dc2e382527ead44c23.md:10]: weren't instructions added in a PR recently?

Comment (ilenia-magoni) [reply curriculum/challenges/english/24-b1-english-for-developers/learn-how-to-describe-places-and-events/66a909dc2e382527ead44c23.md:10]: https://github.com/freeCodeCamp/freeCodeCamp/pull/56472/files it is intended, it is the first application of the new instructions section

Comment (ilenia-magoni) [reply curriculum/challenges/english/24-b1-english-for-developers/learn-how-to-describe-places-and-events/66a909dc2e382527ead44c23.md:10]: cc/ @moT01

Comment (ojeytonwilliams) [reply curriculum/challenges/english/24-b1-english-for-developers/learn-how-to-describe-places-and-events/66a909dc2e382527ead44c23.md:10]: We can probably just make `description` default to an empty string and delete the warning. We already pretend that the type of `description` is `string`, after all, so all the client code is written under the assumption that it is defined.

File: tools/client-plugins/gatsby-source-challenges/create-challenge-nodes.js [hunk 2]
```diff
@@ -7,27 +7,10 @@ function createChallengeNode(
   reporter,
   { isReloading } = { isReloading: false }
 ) {
-  // challengeType 11 is for video challenges (they only have instructions)
-  // challengeType 7 is for certificates (they only have tests)
-  // challengeType 12 is for CodeAlly/CodeRoad challenge
+  if (!challenge.description) challenge.description = '';
+  if (!challenge.instructions) challenge.instructions = '';
+  if (!challenge.seed) challenge.seed = '';
```
Comment (ojeytonwilliams) [inline tools/client-plugins/gatsby-source-challenges/create-challenge-nodes.js:12]: Sorry to be a pain, but this belongs in the `generateChallengeCreator` function in get-challenges. That sets up the challenge, so it's better not to spread the responsibility around.

Also, the output of get-challenges goes through the challenge schema, so we can validate that these properties are defined.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

File: curriculum/get-challenges.js [hunk 3]
```diff
@@ -289,6 +289,11 @@ function generateChallengeCreator(lang, englishPath, i18nPath) {
     challenge.blockType = meta.blockType;
     challenge.hasEditableBoundaries = !!meta.hasEditableBoundaries;
     challenge.order = meta.order;
+
+    if (!challenge.description) challenge.description = '';
+    if (!challenge.instructions) challenge.instructions = '';
+    if (!challenge.seed) challenge.seed = '';
```
Comment (ojeytonwilliams) [inline curriculum/get-challenges.js:295]:
```suggestion
```
Oops. The seed is already checked for (as part of the challenge file validation).

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Review by moT01 — APPROVED** [review]

**Pull request merged**